Add community blog and first blog post

jderochervlk opened the pull request:
- added support for Bluesky for author socials
- updated the community page to include Bluesky and the new community blog
- added a community section to the blog, so now there is Official, Community, and Archived sections.

![image](https://github.com/user-attachments/assets/8faba396-26bf-4eae-b81b-9e745e5924fb)

![image](https://github.com/user-attachments/assets/146e7b9e-d899-4c67-9822-78396bd2fbef)

![image](https://github.com/user-attachments/assets/b81f7e4c-9cda-4e88-8eb0-2b4d0c2ca63c)

fhammerschmidt commented on pages/community/overview.mdx:
```diff
@@ -23,7 +23,8 @@ News are broadcasted on this site's blog, on Bluesky and X. Some extra, less imp
 ## Articles
 
-- [Getting rid of your dead code in ReScript](https://dev.to/zth/getting-rid-of-your-dead-code-in-rescript-3mba)
+- [Community Blog](https://rescript-lang.org/blog/community)
```
Can you make that link relative?

jderochervlk replied:
Should I update the other links to also be relative?

jderochervlk replied:
Done.

jderochervlk commented:
I'm not sure what is causing this error: 
```
Error Preview:
./_blogposts/2020-08-10-bucklescript-is-rebranding.mdx: Unknown href '/blog/bucklescript-8-1-new-syntax' in line 28:3
```

https://github.com/rescript-lang/rescript-lang.org/actions/runs/13638053220/job/38121443178?pr=973#step:6:4361

jderochervlk commented:
> I'm not sure what is causing this error:
> 
> ```
> Error Preview:
> ./_blogposts/2020-08-10-bucklescript-is-rebranding.mdx: Unknown href '/blog/bucklescript-8-1-new-syntax' in line 28:3
> ```
> 
> https://github.com/rescript-lang/rescript-lang.org/actions/runs/13638053220/job/38121443178?pr=973#step:6:4361

Oh, I have to update some links I think. Glad this check is here!

fhammerschmidt commented on _blogposts/community/2025-01-01-what-can-i-do-with-rescript.mdx:
````diff
@@ -0,0 +1,143 @@
+You don't need to write bindings for an entire library, or even for all of a functions arguments. Just write what you need as you go.
+
+Let's take a look at the `format` function from [date-fns](https://date-fns.org/). We can see the [arguments in the docs](https://date-fns.org/v4.1.0/docs/format#arguments), and how it should be imported and used.
+```res
````
`typescript`

fhammerschmidt commented:
Also let's update the date :)

jderochervlk commented:
> The previewImg is still from the release post. Do you have a different one that would fit here?

What kind of images do we want? Do we even need images for every post? We could do the "generic thing from unsplash" like most folks do on other blogging sites. Other languages just seem to omit header images.

![image](https://github.com/user-attachments/assets/ee818bba-61bf-4e0a-8feb-39b2a59a163b)

fhammerschmidt commented:
You can also just remove it, but the 11.1 image is out of place here.

fhammerschmidt commented:
Sorry, it seems to be only "ts", not "typescript". Then syntax highlighting should work.

Also relative links should work from the blogpost? Not sure what broke for you, but please try to make them relative.

jderochervlk commented:
> You can also just remove it, but the 11.1 image is out of place here.

Removed the image to use the default.

> Sorry, it seems to be only "ts", not "typescript". Then syntax highlighting should work.

Fixed.

> Also relative links should work from the blogpost? Not sure what broke for you, but please try to make them relative.

I pushed it up with the relative links, I'll check once the preview build completes.

fhammerschmidt commented:
Found a typo: 
labled arguments -> labeled argmuments

jderochervlk commented:
> Also relative links should work from the blogpost? Not sure what broke for you, but please try to make them relative.

They are currently 404ing.

Fixed the typo.

fhammerschmidt commented:
Seems the vercel deployment works fine.

Only the cloudflare one does not. @cometkim 

As we still publish the vercel one to rescript-lang.org we are fine for now IMO.

cometkim commented:
Yes, it'ok since we don't actually use the Cloudflare one yet. I'll look into it today.
